vmdb/app/controllers/vm_cloud_controller.rb

```diff
@@ -9,29 +9,36 @@ class VmCloudController < ApplicationController
 
   private
 
-  def build_trees_and_accordions
-    @trees   = []
-    @accords = []
-    if role_allows(:feature=>"instances_accord")
-      build_vm_tree(:instances, :instances_tree)  # Build V&T tree
-      @trees.push("instances_tree")
-      @accords.push({:name=>"instances", :title=>"Instances by Provider", :container=>"instances_tree_div"})
-    end
-    if role_allows(:feature=>"images_accord")
-      build_vm_tree(:images, :images_tree)  # Build V&T tree
-      @trees.push("images_tree")
-      @accords.push({:name=>"images", :title=>"Images by Provider", :container=>"images_tree_div"})
-    end
-    if role_allows(:feature=>"instances_filter_accord")
-      build_vm_tree(:filter, :instances_filter_tree) # Build VM filter tree
-      @trees.push("instances_filter_tree")
-      @accords.push({:name=>"instances_filter", :title=>"Instances", :container=>"instances_filter_tree_div"})
-    end
-    if role_allows(:feature=>"images_filter_accord")
-      build_vm_tree(:filter, :images_filter_tree) # Build Template filter tree
-      @trees.push("images_filter_tree")
-      @accords.push({:name=>"images_filter", :title=>"Images", :container=>"images_filter_tree_div"})
-    end
+  def features
+    [
+      ApplicationController::Feature.new("instances_accord",
+                                         :instances,
+                                         "instances",
+                                         :instances_tree,
+                                         "Instances by Provider",
+                                         "instances_tree_div"),
+
+      ApplicationController::Feature.new("images_accord",
+                                         :images,
+                                         "images",
+                                         :images_tree,
+                                         "Images by Provider",
+                                         "images_tree_div"),
+
+      ApplicationController::Feature.new("instances_filter_accord",
+                                         :filter,
+                                         "instances_filter",
+                                         :instances_filter_tree,
+                                         "Instances",
+                                         "instances_filter_tree_div"),
+
+      ApplicationController::Feature.new("images_filter_accord",
+                                         :filter,
+                                         "images_filter",
+                                         :images_filter_tree,
+                                         "Images",
+                                         "images_filter_tree_div")
```

Comment:
@tenderlove So, first of all, how did you restrain yourself on this?

Second, can we use keyword arguments as a phase 1 to give these 6 arguments names?